plog-kafka/src/main/java/com/airbnb/plog/kafka/KafkaHandler.java

```diff
@@ -13,29 +13,31 @@
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
-import java.io.ByteArrayOutputStream;
-import java.util.concurrent.atomic.AtomicLong;
-
 import javax.crypto.Cipher;
 import javax.crypto.spec.SecretKeySpec;
+import java.io.ByteArrayOutputStream;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicLong;
 
 @RequiredArgsConstructor
 @Slf4j
 public final class KafkaHandler extends SimpleChannelInboundHandler<Message> implements Handler {
     private final String defaultTopic;
     private final boolean propagate;
-    private final Producer<byte[], byte[]> producer;
+    private final Producer<Integer, byte[]> producer;
```

Comment:
It might be more general to keep the key as byte[]. Integer can be easily converted to byte[]. 
In addition, it will be useful to let user to specify the key based on the message, and let the partitioner to balance the distribution.